tor/core/inbox.py

```diff
@@ -66,7 +69,12 @@ def process_reply(reply: Comment, cfg: Config) -> None:
         r_body = reply.body.lower()  # cache that thing
 
         if "image transcription" in r_body or is_comment_transcription(reply, cfg):
-            message = i18n["responses"]["general"]["transcript_on_tor_post"]
+            post_link = submission.url
+            sub_name = extract_sub_from_url(submission.url: str) -> str
```

Comment:
The `: str` and `-> str` are type annotations and only valid at the function definition. So `url: str` means that `url` has type string and `-> str` is the same for function return types.
When you use the function, you don't need to provide them, because Python already knows the types from the definition.

```suggestion
            sub_name = extract_sub_from_url(submission.url)
```

The only thing that would be valid here is
```py
sub_name: str = extract_sub_from_url(submission.url)
```
but because we already know that `extract_sub_from_url` returns a `str` from the return type, we don't need to specify the type again.

Also note that in Python type annotations are voluntary, because it's a dynamically typed language. But they can help to detect errors earlier.

Reply:
I see makes sense, thank you!